### PR TITLE
FOR DEMONSTRATION ONLY: test for surface approacher

### PR DESCRIPTION
**Pull request by asalzburger**
This PR is for demonstration only and is not to be merged.

It will be closed after the ACTS workshop, however, the code for the `MeasurementApproacher` is a good starting point for the TPC discussion.

It's a modification of the `PropagationAlgorithm`  which is extended with two actors:
* `MeasurementGenerator`
* `MeasurementApproacher`

This generates hits on given radii:

Screenshot 2020-05-28 at 14 42 19

### CODE REVIEW

**Comment by asalzburger** [comment]
The measurements in this PR area smeared in transverse and longitudinal direction:

```c++
    // Get the output generation result & create smeared space points
    std::vector<Acts::Vector3D> smearedSpacePoints;
    double smearT = 2.;
    double smearL = 5.;
    auto generatorResult = result.template get<Generator::result_type>();
    ACTS_DEBUG("Generator created " << generatorResult.spacepoints.size()
                                    << " measurements.");

    smearedSpacePoints.reserve(generatorResult.spacepoints.size());
    for (auto& pos : generatorResult.spacepoints) {
      Acts::Vector3D spos(pos.x() + smearT * gauss(rnd),
                          pos.y() + smearT * gauss(rnd),
                          pos.z() + smearL * gauss(rnd));
      ACTS_VERBOSE(" --> Measurement at radius = "
                   << Acts::VectorHelpers::perp(pos) << " created.");
      smearedSpacePoints.push_back(spos);
    }
```

Here are the results of the approaches:
residual-x
residual-y
residual-z
residual-3d

**Comment by asalzburger** [comment]
Hey @FabianKlimpel @msmk0 - here's my shot at it, it was actually pretty easy to get it there, we should discuss today.

File: Examples/Algorithms/Propagation/include/ACTFW/Propagation/MeasurementApproacher.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    // Get the position & direction
+    auto position = stepper.position(state.stepping);
+    auto direction = stepper.direction(state.stepping);
+
+    // Let's create a hyperplane at the measurement point
+    using Plane = Eigen::Hyperplane<double, 3>;
+    using Line = Eigen::ParametrizedLine<double, 3>;
+
+    // Make the plane and intersect
+    auto plane = Plane(direction, measurement);
```
Comment (XiaocongAi) [inline Examples/Algorithms/Propagation/include/ACTFW/Propagation/MeasurementApproacher.hpp:60]: I really like those few lines of code! :smile_cat:

**Comment by msmk0** [comment]
@asalzburger As the developer workshop is over, should we close this now?

File: Examples/Algorithms/Propagation/include/ACTFW/Propagation/MeasurementApproacher.hpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Algorithms/Propagation/include/ACTFW/Propagation/MeasurementApproacher.hpp:60]: I will do a test if this actually is more performant than our hand-written planar interjector, if so, we should change to this in any case.

**Comment by asalzburger** [comment]
> @asalzburger As the developer workshop is over, should we close this now?

Yes, the relevant part is already mentioned in an issue.

**Pull request closed without merging**